hugr-core/src/hugr/views/sibling.rs

```diff
@@ -311,29 +283,38 @@ impl<'g, Root: NodeHandle> HugrView for SiblingMut<'g, Root> {
         node == self.root || self.base_hugr().get_parent(node) == Some(self.root)
     }
 
-    fn node_ports(&self, node: Node, dir: Direction) -> Self::NodePorts<'_> {
-        match self.contains_node(node) {
-            true => self.base_hugr().node_ports(node, dir),
-            false => <FlatRegionGraph as PortView>::NodePortOffsets::default().map_into(),
-        }
+    fn node_ports(&self, node: Node, dir: Direction) -> impl Iterator<Item = Port> + Clone {
+        self.base_hugr().node_ports(node, dir)
     }
 
-    fn all_node_ports(&self, node: Node) -> Self::NodePorts<'_> {
-        match self.contains_node(node) {
-            true => self.base_hugr().all_node_ports(node),
-            false => <FlatRegionGraph as PortView>::NodePortOffsets::default().map_into(),
-        }
+    fn all_node_ports(&self, node: Node) -> impl Iterator<Item = Port> + Clone {
+        self.base_hugr().all_node_ports(node)
     }
 
-    fn linked_ports(&self, node: Node, port: impl Into<Port>) -> Self::PortLinks<'_> {
+    fn linked_ports(
+        &self,
+        node: Node,
+        port: impl Into<Port>,
+    ) -> impl Iterator<Item = (Node, Port)> + Clone {
         // Need to filter only to links inside the sibling graph
-        SiblingGraph::<'_, Node>::new_unchecked(self.hugr, self.root)
-            .linked_ports(node, port)
-            .collect::<Vec<_>>()
-            .into_iter()
+        let graph = self.graph();
+        let port = graph
+            .port_index(node.pg_index(), port.into().pg_offset())
+            .unwrap();
+        self.hugr
+            .graph
+            .port_links(port)
+            .map(|(_, link)| {
+                let graph = self.graph();
+                let port: PortIndex = link.into();
+                let node: Node = graph.port_node(port).unwrap().into();
+                let offset: Port = graph.port_offset(port).unwrap().into();
+                (node, offset)
+            })
+            .filter(|(n, _)| self.contains_node(*n))
     }
 
-    fn node_connections(&self, node: Node, other: Node) -> Self::NodeConnections<'_> {
+    fn node_connections(&self, node: Node, other: Node) -> impl Iterator<Item = [Port; 2]> + Clone {
         // Need to filter only to connections inside the sibling graph
         SiblingGraph::<'_, Node>::new_unchecked(self.hugr, self.root)
```

Comment:
would some of these methods also benefit from the portgraph based implementation seen in `linked_ports`